.gitlab-ci.yml

```diff
@@ -76,6 +76,7 @@ build:
       -DCMAKE_BUILD_TYPE=Release 
       -DCMAKE_CXX_FLAGS=-w 
       -DCMAKE_CUDA_FLAGS=-w 
+      -DCMAKE_CUDA_ARCHITECTURES=86
```

Comment:
Well, what happens without this setting? I know that the A4000 is supposed to support compute capability 8.6. But I still suspect that this is where things go wrong in the test. That somehow the created binary is not compatible with the card.

Reply:
I was getting the same error without this flag (i.e. the default CI configuration), that's why @benjaminhuth started investigating.

Reply:
Mmh what I try at the moment (not yet finished) is to rebuild the docker image with `-DCMAKE_CUDA_ARCHITECTURES=75;86`, because there are some dependencies (cugraph, ...) which where only built with `-DCMAKE_CUDA_ARCHITECTURES=75`
Not sure if this helps though

Reply:
The CMake configuration/build says the following in that job:

```
...
-- Automatic GPU detection failed. Building for common architectures.
-- Autodetected CUDA architecture(s): 3.5;5.0;5.2;6.0;6.1;7.0;7.5;8.0;8.6;8.6+PTX
-- Added CUDA NVCC flags for: -gencode;arch=compute_35,code=sm_35;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_52,code=sm_52;-
gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-
gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_86,code=compute_86
...
```

So setting `CMAKE_CUDA_ARCHITECTURES` may not actually be effective. :confused:

I'm out of my depth a little bit here. But do you see how the 8.6 version is set up a little differently from the rest?
  - ` -gencode;arch=compute_35,code=sm_35`;
  - `-gencode;arch=compute_50,code=sm_50`;
  - `-gencode;arch=compute_52,code=sm_52`;
  - `-gencode;arch=compute_60,code=sm_60`;
  - `-gencode;arch=compute_61,code=sm_61`;
  - `-gencode;arch=compute_70,code=sm_70`;
  - `-gencode;arch=compute_75,code=sm_75`;
  - `-gencode;arch=compute_80,code=sm_80`;
  - `-gencode;arch=compute_86,code=sm_86`;
  - `-gencode;arch=compute_86,code=compute_86`.

Though I don't know why, or what this would result in... :thinking:

Reply:
Okay, the rebuild of the Docker image has fixed the problem :)